src/ast/passes/macro_expansion.cpp

```diff
@@ -61,12 +61,16 @@ class MacroExpander : public Visitor<MacroExpander> {
   void visit(Variable &var);
   void visit(Map &map);
   void visit(Expression &expr);
+  // We can't add extra params with default values to the standard `visit`
+  // because it then becomes ambiguous
```

Comment:
This exposition is useful to justify what a specific design was chosen, but I feel like it adds noise for someone trying to understand the design. If this overload doesn't work (apologies, I thought did), I would just use something like `rewrite` or `expand` so it doesn't need any explaining and isn't conflicting with `visit` at all.

Reply:
Ack.